app/components/cache-classification.js

```diff
@@ -0,0 +1,67 @@
+import apiClient from 'panoptes-client/lib/api-client';
+
+class CacheClassification {
+  constructor() {
+    this.state = {
+      cachedClassification: null,
+    };
+
+    window.cachedClassification = this.state.cachedClassification;
+  }
+
+  create() {
+    const cachedClassification = apiClient.type('classifications').create({
+      annotations: [],
+      id: 'CACHED_CLASSIFICATION_DO_NOT_SAVE',
+    });
+
+    this.state = { cachedClassification };
+    return this.state.cachedClassification;
+  }
+
+  update(newAnnotation) {
+    let cachedClassification;
+    const taskKey = newAnnotation.task;
+
+    if (this.state.cachedClassification === null) {
+      cachedClassification = this.create();
+    } else {
+      cachedClassification = this.state.cachedClassification;
+    }
+    const cachedAnnotation = this.isAnnotationCached(taskKey);
+
+    if (cachedAnnotation) {
+      const index = cachedClassification.annotations.findIndex(
+        (annotation) => annotation.task === taskKey);
+
+      if (index > -1) {
+        cachedClassification.annotations.splice(index, 1);
+      }
+    }
+    cachedClassification.annotations.push(newAnnotation);
+
+    cachedClassification.update('annotations');
+  }
+
+  delete() {
+    this.state = { cachedClassification: null };
+  }
```

Comment:
we think the apiClient is keeping a cache, which causes annotations given for a previous subject to sometimes pre-populate for a different subject. though since it's not saving the resource, `.delete()` isn't working. however, I think adding...
```
this.state.cachedClassification.update('annotations');
```
after the state is updated to null, will fix the issue?